### PR TITLE
feat(Transactions): add loading spinner

### LINKED ISSUE #2197: [Feature] Add pagination or loading icon on transactions window.
### Feature description

Hi, I have a lot of transactions on transactions window. I see it's a little slow to load transactions. I think this could be solved by adding pagination or a loading icon. See the video:

https://user-images.githubusercontent.com/1178856/224490183-4fae7224-6186-400f-9a30-94175ea29cda.mp4

### Describe the solution

I think this could be solved by adding pagination or a loading icon. 

### Describe alternatives

_No response_

### Additional context

_No response_

### Are you working on this?

No

### PR DESCRIPTION
**Pull request by lujakob**
### Describe the changes you have made in this PR

Add a loading spinner to transactions page. This is a first step to improve UX for long lists of data.

### Link this PR to an issue [optional]

Fixes #2197

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### How has this been tested?

Manually

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Review by escapedcat — APPROVED** [review]
utack

**Review by reneaaron — APPROVED** [review]
tACK

File: src/app/screens/Transactions/index.tsx [hunk 1]
```diff
@@ -15,6 +16,7 @@ function Transactions() {
 
   const { settings, getFormattedFiat } = useSettings();
   const [transactions, setTransactions] = useState<Transaction[]>([]);
+  const [transactionsLoading, setTransactionsLoading] = useState<boolean>(true);
```
Comment (reneaaron) [inline src/app/screens/Transactions/index.tsx:19]: I think the default state should be `false`and `setTransactionsLoading(true)` should be called in `fetchData`. That way the loading state would also appear with repeated loading.

**Review by reneaaron — CHANGES_REQUESTED** [review]
Thanks @lujakob! 

Some little nit, other than that: LGTM!

File: src/app/screens/Transactions/index.tsx (hunk 1 above)
Comment (lujakob) [reply src/app/screens/Transactions/index.tsx]: I updated accordingly.

**Review by reneaaron — APPROVED** [review]
tACK

**Pull request merged**